AjobK/Athena
# 213 admin ban user

closes #213 

<br />

 **What is it supposed to do**  
Make admins able to ban an account until a certain time (banned_by which stores admin id and banned_until which stores a datetime)

In addition to its original requirements, I also added a filter to the getProfile function so that it only returns users that aren't banned, and when a user is banned his current posts are hidden, and he can't log in.

<br />

 **How can we test your branch**  
Explain how can test your branch as specifically as possible. Mention the page, button, etc. (For back-end route, function, etc.)
You can test the branch by following these steps:
- reset your database
- run the backend and frontend
- login as admin/ moderator or head-admin
- go to the user profile 
- when you hover over the users banner, a ban icon should be displayed
- if you click on the banner, a popup should be displayed 
- fill in the amount of days a user should be banned and the reason
- click save
- if you refresh the page, the user's account shouldn't be displayed
- try to login as the user you (you shouldn't be able to)
- check if there are any posts made by that user

<br />

**General checklist**
- [x] Tested happy flow.
- [x] Tested unhappy flow.
- [x] No unexpected exceptions.
- [x] No code with high complexity. (Big O)
- [x] EOF newline.
- [x] No floating comma's and spaces.
- [x] No unexpected/unhandled errors.
- [x] Lines no longer than 'guideline' in IDE.
- [x] Correct indentation.
- [x] No usage of legacy functionalities.
- [x] (JS) Curly braces on the sameline as reserved keywords *(`if () {` instead of `{` on line below)*
- [x] (JS) No double quotes.
- [x] (JS) Single newline above unique code structures and reserved keywords.
- [x] (JS) Variables bundled where possible.
- [x] (JS) No semi-colons except where they have a functional necessity, such as for-loops.

<br />

**If relevant, front-end checklist**
- [x] Responsive. (Desktop all the way to mobile 320px)
- [x] Using SCSS variables where relevant.
- [x] Relevant class names
- [x] No modified global SCSS properties, except if necessary and fully tested the impact.
- [ ] Checked [caniuse](https://caniuse.com) for more modern CSS properties.

<br />

**If relevant, back-end checklist**
- [x] No unhandled promise rejections.
- [x] Seeders still functional.
- [x] No primary-key violations.
- [x] Normalization, atleast [3NF](https://en.wikipedia.org/wiki/Database_normalization).
- [x] Used guards (middleware) where necessary and tested if route is guarded.

<br />

**If relevant, test these browsers**
 
---

**PR Rules**  
*1. Add relevant labels.*  
*2. Select atleast two reviewers.*  
*3. Assign all reviewers allowed to merge the branch, including yourself.*  
*4. If relevant to the sprint, put related issue in 'Awaiting Approval' column on project board.*  
*5. Add to the corresponding [milestone](https://github.com/AjobK/Seaqull/milestones).*
